Source/SwiftLintFramework/Extensions/Configuration+Parsing.swift

```diff
@@ -0,0 +1,128 @@
+//
+//  Configuration+Parsing.swift
+//  SwiftLint
+//
+//  Created by JP Simard on 7/17/17.
+//  Copyright © 2017 Realm. All rights reserved.
+//
+
+import Foundation
+
+extension Configuration {
+    internal enum Key: String {
+        case cachePath = "cache_path"
+        case disabledRules = "disabled_rules"
+        case enabledRules = "enabled_rules" // deprecated in favor of optInRules
+        case excluded = "excluded"
+        case included = "included"
+        case optInRules = "opt_in_rules"
+        case reporter = "reporter"
+        case swiftlintVersion = "swiftlint_version"
+        case useNestedConfigs = "use_nested_configs" // deprecated
+        case warningThreshold = "warning_threshold"
+        case whitelistRules = "whitelist_rules"
+    }
+
```

Comment:
yeah, I was thinking of doing that in a fresh PR to reduce the scope of this and #1687.